bigquery/src/main/java/org/apache/zeppelin/bigquery/bigQueryInterpreter.java

```diff
@@ -0,0 +1,340 @@
+    while (!job.getStatus().getState().equals("DONE")) {
+      System.out.println("Job is " 
+          + job.getStatus().getState() 
+          + " waiting " + interval + " milliseconds...");
+      Thread.sleep(interval);
+      job = request.execute();
+    }
+    return job;
+  }
+
+  //Function to page through the results of an arbitrary bigQuery request
+  public static <T extends GenericJson> Iterator<T> getPages(
+      final BigqueryRequest<T> requestTemplate) {
+    class PageIterator implements Iterator<T> {
+      private BigqueryRequest<T> request;
+      private boolean hasNext = true;
+      public PageIterator(final BigqueryRequest<T> requestTemplate) {
+        this.request = requestTemplate;
+      }
+      public boolean hasNext() {
+        return hasNext;
+      }
+      public T next() {
+        if (!hasNext) {
+          throw new NoSuchElementException();
+        }
+        try {
+          T response = request.execute();
+          if (response.containsKey("pageToken")) {
+            request = request.set("pageToken", response.get("pageToken"));
+          } 
+          else {
+            hasNext = false;
+          }
+          return response;
+        } 
+        catch (IOException e) {
+          e.printStackTrace();
+          return null;
+        }
+      }
+
+      public void remove() {
+        this.next();
+      }
+    }
+
+    return new PageIterator(requestTemplate);
+  }
+  
+  //Function to call bigQuery to run SQL and return results to the Interpreter for output
+  private InterpreterResult executeSql(String sql) {
+    int counter = 0;
+    StringBuilder finalmessage = null;
+    finalmessage = new StringBuilder("%table ");
+    String projId = getProperty(PROJECT_ID);
+    long wTime = Long.parseLong(getProperty(WAIT_TIME));
+    long maxRows = Long.parseLong(getProperty(MAX_ROWS));
+    Iterator<GetQueryResultsResponse> pages = run(sql, projId, wTime, maxRows);
+    try {
+      while (pages.hasNext()) {
+        finalmessage.append(printRows(pages.next()));
+      }
+      return new InterpreterResult(Code.SUCCESS, finalmessage.toString());
+    }
+    catch ( NullPointerException ex ) {
+      return new InterpreterResult(Code.ERROR, ex.getMessage());
+    }
+  }
+
+  //Function to run the SQL on bigQuery service
+  public static Iterator<GetQueryResultsResponse> run(final String queryString, 
+    final String projId, final long wTime, final long maxRows) {
+    try {
+      QueryResponse query = service.jobs().query(
+          projId,
+          new QueryRequest().setTimeoutMs(wTime).setQuery(queryString).setMaxResults(maxRows))
+          .execute();
+      jobId = query.getJobReference().getJobId();
+      projectId = query.getJobReference().getProjectId();
+      GetQueryResults getRequest = service.jobs().getQueryResults(
+          projectId,
+          jobId);
+      return getPages(getRequest);
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+      return null;
```

Comment:
We use Log4j for logging in Zeppelin, it would be better to use it and replace `printStackTrace()` with `log.error()`